### PR TITLE
Support configurable backend gyro rates on Invensense sensors

### PR DESCRIPTION
**Pull request by andyp1per**
This PR allows the fast sampling backend rate to be raised through the configuration parameter ```INS_GYRO_RATE```
0 = 1Khz backend rate
1 = 2Khz
2 = 4Khz
It is a more general version of #14389 and makes the default on F7 and H7 2Khz
Gyros and accels still share the same backend rate

Renames get_sample_rate_hz() -> get_loop_rate_hz() since that is what it is.

After extensive testing:

- Using this on F4 processors is fine as long as you don't use double-notches. If you use 3 double-notches you are using the CPU load equivalent of 12 notches and its right on the edge. Keeping the total notches at 3-4 for 2k backend seems an acceptable compromise (so two harmonics with double notches or 3 harmonics with single notches).
- F7 is fine up to 4Khz but the EKF does not like the amount of noise coming through at 4Khz (no averaging at this rate). It might be worth making the 188Hz accel filter more aggressive by converting it to a two-pole filter

### CODE REVIEW

**Comment by andyp1per** [comment]
I flew some brief flights on my min-quad with 4k gyro rates using this PR. There was no discernible problem with performance:

image

And post-filter noise looked nice and low (note the 4k sample rate):

image

**Comment by andyp1per** [comment]
I've been flying with this for a couple of weeks now and it works well. I think it needs to be configurable because running with faster gyros tends to introduce some noise which might be unwelcome on noisier builds - but on clean builds it allows you to go much higher with filtering frequencies, particularly the harmonic notch.

File: libraries/AP_InertialSensor/AP_InertialSensor.cpp [hunk 1]
```diff
@@ -493,6 +501,14 @@ const AP_Param::GroupInfo AP_InertialSensor::var_info[] = {
     // @Path: ../Filter/HarmonicNotchFilter.cpp
     AP_SUBGROUPINFO(_harmonic_notch_filter, "HNTCH_",  41, AP_InertialSensor, HarmonicNotchFilterParams),
 
+    // @Param: GYRO_RATE
+    // @DisplayName: Fast sampling gyro rate for enabled IMUs
+    // @Description: Gyro rate for IMUs with fast sampling enabled. If the sensor does not support the selected rate the next highest supported rate will be used.
+    // @User: Advanced
+    // @Values: 0:1 kHz,1:2 kHz,3:4 kHz
```
Comment (rmackay9) [inline libraries/AP_InertialSensor/AP_InertialSensor.cpp:508]: We should probably have a Units for kHz?  Not a blocker

Comment (rmackay9) [reply libraries/AP_InertialSensor/AP_InertialSensor.cpp:508]: Ah, it's a selection sorry..

File: libraries/AP_InertialSensor/AP_InertialSensor.cpp [hunk 2]
```diff
@@ -493,6 +501,14 @@ const AP_Param::GroupInfo AP_InertialSensor::var_info[] = {
     // @Path: ../Filter/HarmonicNotchFilter.cpp
     AP_SUBGROUPINFO(_harmonic_notch_filter, "HNTCH_",  41, AP_InertialSensor, HarmonicNotchFilterParams),
 
+    // @Param: GYRO_RATE
+    // @DisplayName: Fast sampling gyro rate for enabled IMUs
```
Comment (rmackay9) [inline libraries/AP_InertialSensor/AP_InertialSensor.cpp:505]: Maybe let's change this to "Gyro rate for IMUs with Fast Sampling enabled"

**Comment by rmackay9** [comment]
As discussed on the dev call, I'm concerned that users will be confused between the FAST_SAMPLING bitmask parameter and the new GYRO_RATE parameter.  To alleviate this we agreed:

1. add a wiki document to tell users about how to set their IMU and Gyro rate using the IMU_FAST_SAMPLING and GYRO_RATE parameter
2. add a banner to tell users what update rate their IMU and/or Gyro are running at.

I'm happy for this to go in now, txs!

**Comment by andyp1per** [comment]
@rmackay9 ok?
image

File: libraries/AP_InertialSensor/AP_InertialSensor.cpp (hunk 2 above)
Comment (andyp1per) [reply libraries/AP_InertialSensor/AP_InertialSensor.cpp:505]: Fixed

**Comment by andyp1per** [comment]
I've made the doc and banner changes. No functional changes so good to merge when you are happy @rmackay9

File: libraries/AP_InertialSensor/AP_InertialSensor.cpp [hunk 3]
```diff
@@ -979,6 +995,16 @@ AP_InertialSensor::init_gyro()
     _save_gyro_calibration();
 }
 
+// output GCS startup messages
+bool AP_InertialSensor::get_output_banner(uint8_t backend_id, char* banner, uint8_t banner_len)
```
Comment (rmackay9) [inline libraries/AP_InertialSensor/AP_InertialSensor.cpp:999]: It's not a blocker but for the future this could be made "const" I think.

**Pull request merged**